### PR TITLE
SPCTR-2121 - New : Core Block - Icon Block

### PR DESCRIPTION
**Pull request by adi3890**
### Description
New Core Block - Icon Block
Icon Block will be 7th core block of Spectra
Allows user to choose around 1800 icons from FontAwesome Family

[Demo](http://spectra-dev.vgax7ebrln-wg96glqzy3oy.p.temp-site.link/icon-block/)

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: includes/blocks/icon/block.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+$block_slug = 'uagb/icon';
+$block_data = array(
+	'doc'              => 'icon',
+	'slug'             => '',
+	'admin_categories' => array( 'core' ),
+	'link'             => 'icon',
+	'title'            => __( 'Icon', 'ultimate-addons-for-gutenberg' ),
+	'description'      => __( 'Add Icons to you page.', 'ultimate-addons-for-gutenberg' ),
```
Comment (RajkiranBagal) [inline includes/blocks/icon/block.php:17]:
```suggestion
	'description'      => __( 'Add Icons to your page.', 'ultimate-addons-for-gutenberg' ),
```

File: src/blocks/icon/render.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		block_id,
+	} = attributes;
+
+	const iconSvg = icon ? icon : 'circle-check';
+	const iconHtml = (
+			( renderSVG( iconSvg, setAttributes ) )
+		)
+
+	const blockProps = useBlockProps( {
+		className: classnames( `uagb-block-${ block_id }` )
```
Comment (RajkiranBagal) [inline src/blocks/icon/render.js:23]: if classnames is not needed here then we can remove

File: src/blocks/icon/save.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+				target = { targetVal }
+			>
+				{ renderSVG( icon ? icon : 'circle-check' ) }
+			</a> )
+	} else {
+		iconHtml = ( renderSVG( icon ? icon : 'circle-check' ) )
+	}
+
+	const blockProps = useBlockProps.save( {
+		className: classnames( `uagb-block-${ block_id }` )
```
Comment (RajkiranBagal) [inline src/blocks/icon/save.js:43]: if classnames is not needed here then we can remove

File: src/blocks/icon/settings.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Block Icon : Settings.
+ */
+import React from 'react';
```
Comment (RajkiranBagal) [inline src/blocks/icon/settings.js:4]: If not needed then remove

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: includes/blocks/icon/block.php (hunk 1 above)
Comment (adi3890) [reply includes/blocks/icon/block.php:17]: Replaced with **Add beautiful icons to your website.**
Keeping it same in both JS and PHP

File: src/blocks/icon/render.js (hunk 2 above)
Comment (adi3890) [reply src/blocks/icon/render.js]: removed

File: src/blocks/icon/save.js (hunk 3 above)
Comment (adi3890) [reply src/blocks/icon/save.js]: removed

File: src/blocks/icon/settings.js (hunk 4 above)
Comment (adi3890) [reply src/blocks/icon/settings.js]: replaced with memo from @wordpress.element

**Review by rahulvarma722 — APPROVED** [review]

**Pull request merged**